### PR TITLE
Some util functions to support bpfix

### PR DESCRIPTION
**Pull request by mathjeff**
Change-Id: If53f696fbe937e007b902434f62d0d92435846dd

### CODE REVIEW

File: parser/ast.go [hunk 1]
```diff
@@ -122,6 +122,35 @@ type Expression interface {
 	Eval() Expression
 }
 
+func ExpressionsHaveSameMeaning(a Expression, b Expression) (equal bool, err error) {
```
Comment (colincross) [inline parser/ast.go]: What's the meaning of "Meaning"?  This function at least needs a godoc comment, but probably also a different name.  ExpressionsAreSame seems clearer to me, or ExpressionsAreIdentical.

File: parser/ast.go [hunk 2]
```diff
@@ -122,6 +122,35 @@ type Expression interface {
...
+		return false, nil
+	}
+	right, err := hackyFingerprint(b)
+	if err != nil {
+		return false, nil
+	}
+	areEqual := string(left) == string(right)
+	return areEqual, nil
+}
+func hackyFingerprint(expression Expression) (fingerprint []byte, err error) {
```
Comment (colincross) [inline parser/ast.go]: blank line between functions.

File: parser/ast.go [hunk 3]
```diff
@@ -233,6 +262,26 @@ func (x *Map) String() string {
...
+}
+func (x *Map) getPropertyImpl(name string) (Property *Property, found bool, index int) {
+	for i, prop := range x.Properties {
+		if prop.Name == name {
+			return prop, true, i
+		}
+	}
+	return nil, false, -1
+}
+func (x *Map) Remove(propertyName string) (removed bool) {
```
Comment (colincross) [inline parser/ast.go]: RemoveProperty?

File: parser/ast.go [hunk 4]
```diff
@@ -233,6 +262,26 @@ func (x *Map) String() string {
 
 func (x *Map) Type() Type { return MapType }
 
+func (x *Map) GetProperty(name string) (Property *Property, found bool) {
```
Comment (colincross) [inline parser/ast.go]: Generally all exported functions should have at least a simple godoc comment, although we are not always good at following that rule.

**Comment by mathjeff** [comment]
Ping

**Comment by mathjeff** [comment]
Reminder that this is a prerequisite for the current implementation of bpfix

**Comment by mathjeff** [comment]
Thanks! :)

**Pull request merged**